### PR TITLE
4GB RAM requirement for building from source

### PR DESCRIPTION
**Pull request by jseldess**
Based on https://github.com/cockroachdb/docs/issues/421

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/7509)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -16,7 +16,7 @@
 	```
 
 3.  Run `make build`, `make test`, or anything else our Makefile
-	offers. Note that the first time you run `make`, it can take some
+	offers. Note that at least 4GB of RAM is required to build from source. Also, the first time you run `make`, it can take some
```
Comment (tamird) [inline CONTRIBUTING.md]: please consistently wrap this at 80 characters

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -85,70 +85,67 @@ To add or update a go dependency:
...
+* the [net/trace](https://godoc.org/golang.org/x/net/trace) endpoint at 
+  `/debug/requests`.  It has a breakdown of the recent traced requests, in 
+  particularly slow ones. Two families are traced: `node` and `coord`, the 
+  former (and likely more interesting one) containing what happens inside of 
+  `Node`/`Store`/`Replica` and the other inside of the coordinator 
+  (`TxnCoordSender`).
+* [pprof](https://golang.org/pkg/net/http/pprof/) gives us (among other things) 
+  heap and cpu profiles; [this golang blog post]
+  (http://blog.golang.org/profiling-go-programs) explains it extremely well and 
+  [this one by Dmitry Vuykov](https://software.intel.com/en-us/blogs/2014/05/10/debugging-performance-issues-in-go-programs)
```
Comment (tamird) [inline CONTRIBUTING.md]: i think you can break this one after the `]`

**Comment by tamird** [comment]
LGTM

**Comment by jseldess** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[CONTRIBUTING.md, line 19 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7509#-KLNQdaITHZLRTyZReZs-r1-19:-KLNhs6XwqQf5KVOb-47:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/b1eaf37440c1572e8e470713c05d0ad16169f64b/CONTRIBUTING.md#L19)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> please consistently wrap this at 80 characters
</details>
Done.

---

*[CONTRIBUTING.md, line 128 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7509#-KLNQdaITHZLRTyZReZs-r3-128:-KLNhskD5D2f0sNxAPtL:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/990e72353731cfadf81623978f44ffb863eebf4a/CONTRIBUTING.md#L128)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> i think you can break this one after the `]`
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7509)*

**Pull request merged**